### DISCUSSION TITLE
@mention roles via discord webhook

### DISCUSSION THREAD

**Discussion by michaelkr1** [containrrr/shoutrrr#295] [Ideas]
Would be awesome to be able to @mention users or groups for the Discord webhook (I assume this could extend into other notification URLs for Teams and Slack).

I use Watchtower but I didn't know whether to open the discussion here or there.

**Comment by piksel** [comment]
It should be possible to mention by simply adding the `@user`s to the message (by changing the template in watchtowers case).

**Reply by michaelkr1** [reply]
Unfortunately it doesn't ping because discord doesn't allow @mentions within embeds (according to: https://stackoverflow.com/a/70583258)

It would need to be:
@Mention
<watchtower notification embedded messages>

**Reply by michaelkr1** [reply]
To clarify on this point a little. You can add your name/role to embeds using ID's (https://docs.discord.club/embedg/reference/mentions) however it doesn't actually mention the name or role and only prints it. For it to actually ping the user/role, it needs to be outside of the embed.

**Reply by piksel** [reply]
Yeah, that seems to be true. It should work by putting the mentions in the `content` though... Perhaps support could be added by adding a new prop called `pings` or similar, and passing them in the `content`.

**Reply by NonaSuomy** [reply]
If you are using multiple options, URLs, and Email notifications how would you add the discord 18-digit user_id <@XXXXXXXXXXXXXXXXXX> to the notification template but only dump it for the discord notification and not the email notification?

**Reply by piksel** [reply]
> If you are using multiple options, URLs, and Email notifications how would you add the discord 18-digit user_id <@XXXXXXXXXXXXXXXXXX> to the notification template but only dump it for the discord notification and not the email notification?
> 

It wouldn't be part of the template, but instead a query param added to the `discord:` service URL.
